cli/pkg/edgeclient/types.go

```diff
@@ -0,0 +1,42 @@
+/*
+Copyright 2024. Open Data Hub Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package edgeclient
+
+type ModelImageStatus int
+
+func (s ModelImageStatus) String() string {
+	return [...]string{"Unknown", "Needs Sync", "Synced", "Building", "Live", "Failed"}[s]
+}
```

Comment:
For my own understanding, do you have any links to docs on this kind of construct, or a name of a pattern or something that I could search for? I don't remember having ever encountered it before (It has been more years since I last used Go than I actually used it for, so I might just have forgotten).

Reply:
> For my own understanding, do you have any links to docs on this kind of construct, or a name of a pattern or something that I could search for? I don't remember having ever encountered it before (It has been more years since I last used Go than I actually used it for, so I might just have forgotten).

@grdryn So, this was supposed to be an enum, Go's version of it anyway, but the `ModelImageStatus` was not the correct reference to that since when I split it, I forgot to introduce the constants. 

Here's a good article about [implementing Enum's in Go](https://dizzy.zone/2024/01/26/Enums-in-Go/).

Reply:
I had a feeling it was something around enums, and Go's lack of useful features. 👍